### PR TITLE
refactor!: remove obsolete BoundIndices enum values

### LINKED ISSUE #190: Remove eLOC_0 et al. from BoundParametersIndices enum
They are superseded by the `eBoundLoc{0,1}, eBoundPhi, ...` enum values.

### PR DESCRIPTION
**Pull request by msmk0**
This removes the following deprecated values from the `BoundIndices` enum
```
eLOC_0
eLOC_1   
eLOC_R   
eLOC_PHI 
eLOC_RPHI
eLOC_Z   
eLOC_X   
eLOC_Y   
eLOC_D0  
eLOC_Z0  
ePHI    
eTHETA    
eQOP
eT
```
and replaces them with their correct `eBound...` counterpart.

Fixes #190.

### CODE REVIEW

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectoryWriter.hpp [hunk 1]
```diff
@@ -126,14 +126,14 @@ class RootTrajectoryWriter final : public WriterT<TrajectoryContainer> {
...
   bool m_hasFittedParams;        ///< if the track has fitted parameter
-  float m_eLOC0_fit{-99.};       ///< fitted parameter eLOC_0
-  float m_eLOC1_fit{-99.};       ///< fitted parameter eLOC_1
+  float m_eLOC0_fit{-99.};       ///< fitted parameter eBoundLoc0
+  float m_eLOC1_fit{-99.};       ///< fitted parameter eBoundLoc1
   float m_ePHI_fit{-99.};        ///< fitted parameter ePHI
   float m_eTHETA_fit{-99.};      ///< fitted parameter eTHETA
   float m_eQOP_fit{-99.};        ///< fitted parameter eQOP
   float m_eT_fit{-99.};          ///< fitted parameter eT
   float m_err_eLOC0_fit{-99.};   ///< fitted parameter eLOC_-99.err
```
Comment (XiaocongAi) [inline Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectoryWriter.hpp:135]: Ha, could you help fix the comment here?

File: Core/include/Acts/Surfaces/detail/DiscSurface.ipp [hunk 2]
```diff
@@ -44,25 +44,25 @@ inline void DiscSurface::initJacobianToGlobal(const GeometryContext& gctx,
...
-  jacobian.block<3, 1>(0, eLOC_0) =
+  jacobian.block<3, 1>(0, eBoundLoc0) =
       lcos_phi * rframe.block<3, 1>(0, 0) + lsin_phi * rframe.block<3, 1>(0, 1);
-  jacobian.block<3, 1>(0, eLOC_1) =
+  jacobian.block<3, 1>(0, eBoundLoc1) =
       lrad * (lcos_phi * rframe.block<3, 1>(0, 1) -
               lsin_phi * rframe.block<3, 1>(0, 0));
   // the time component
-  jacobian(3, eT) = 1;
+  jacobian(3, eBoundTime) = 1;
```
Comment (XiaocongAi) [inline Core/include/Acts/Surfaces/detail/DiscSurface.ipp:58]: It's not relevant to this PR, but I just wonder why we use explicit index for the free parameters indices while we use the enum for the bound parameters here and elsewhere?

Comment (msmk0) [reply Core/include/Acts/Surfaces/detail/DiscSurface.ipp:58]: There is no good reason. This should be fixed. Can you open an issue so we can review the usage of bare indices after `v1.0.0`?

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectoryWriter.hpp (hunk 1 above)
Comment (msmk0) [reply Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectoryWriter.hpp]: Done.

File: Core/include/Acts/Surfaces/detail/DiscSurface.ipp (hunk 2 above)
Comment (XiaocongAi) [reply Core/include/Acts/Surfaces/detail/DiscSurface.ipp:58]: Sure, will do it.

**Review by XiaocongAi — APPROVED** [review]
It's nice to have the redundancy removed! Approved.

**Pull request merged**